### DISCUSSION TITLE
overwriteRequestBody for array

### DISCUSSION THREAD

**Discussion by schelv** [apideck-libraries/portman#195] [Q&A] [answered]
Hi I want to overwrite the body of a request.
The documentation mentions keys and values.
I guess this works well for objects.

However I want the body to be an array (of strings) instead of an object.
Is this possible?

**Comment by thim81** [comment, marked as answer]
hi @schelv 

It should be possible. Example:
```
{
      key: 'websites',
      value: [
        {
          url: 'http://widget.biz',
          id: '5678',
          type: 'tertiary'
        }
      ],
      overwrite: true
    }
```

Which will overwrite the `websites` array, with the newly provided array.

(note: I havent tested this myself, but I recall that we did a lot of work to support a wide range of overwrite cases)

**Comment by schelv** [comment]
I should have given an example =)

**example:**
You want to replace the old request body:
```
["this is a string", "this is another string"]
```
with the new request body:
```
["this is a replacement string", "this is another replacement string"]
```

**Comment by thim81** [comment]
You could do like so:

```
{
      key: '[0]',
      value: "this is a replacement string",
      overwrite: true
    },{
      key: '[1]',
      value: "this is another replacement string",
      overwrite: true
    }
```

**Comment by thim81** [comment]
@schelv Did it work for you? 
If yes, feel free to close the issue.
If no, let us know what you struggle with, so we can help you.

**Comment by schelv** [comment]
```
"overwriteRequestBody": [	  
	{
      "key": "[0]",
      "value": "this is a replacement string",
      "overwrite": true
    },{
      "key": "[1]",
      "value": "this is another replacement string",
      "overwrite": true
    }
]
```
With this the request body becomes `[
    "this is a replacement string"
]`

So only one is replaced, so it kind of works. 😅
